cachi2/core/package_managers/cargo/main.py

```diff
@@ -122,26 +123,25 @@ def _resolve_cargo_package(
 
     components = chain((main_component,), deps_components)
 
-    return components
+    return components, _swap_sources_directory_for_subsitution_slot(config_template)
 
 
-def _use_vendored_sources(package_dir: RootedPath) -> ProjectFile:
+def _swap_sources_directory_for_subsitution_slot(template: str) -> dict:
+    toml_template = tomlkit.parse(template).value
+    # Absolute path has to be replaced with relative path for sources relocation to work:
+    toml_template["source"]["vendored-sources"]["directory"] = "${output_dir}/deps/cargo"
+    # A correct output_dir value will be supplied by cachi2 during a later stage.
+    return toml_template
+
+
+def _use_vendored_sources(package_dir: RootedPath, config_template: dict) -> ProjectFile:
     """Make sure cargo will use the vendored sources when building the project."""
     cargo_config = package_dir.join_within_root(".cargo/config.toml")
     cargo_config.path.parent.mkdir(parents=True, exist_ok=True)
     cargo_config.path.touch(exist_ok=True)
```

Comment:
So the intent of this patch is to only update `.cargo/config.toml` with the `inject-files` command, but this pre-existing hunk touches the file if missing, thus making premature changes to the repo, so that needs to be addressed as well so that the file is ONLY created in `inject-files`.

Reply:
Thank you for catching this!